compiler/commands.nim

```diff
@@ -628,6 +628,8 @@ proc processSwitch*(switch, arg: string, pass: TCmdLinePass, info: TLineInfo;
       defineSymbol(conf.symbols, "executable")
       defineSymbol(conf.symbols, "consoleapp")
     of "lib":
+      # should that apply also for other `arg` options?
+      conf.setNoteDefaults(warnUnusedImportX, false) # similar to issue #13218
```

Comment:
Inconsistent. Why does a library build all of a sudden not care about unused imports anymore?

Reply:
for typical use like lib/nimrtl.nim which defines:
```
import
  parseutils, strutils, parseopt, parsecfg, strtabs, unicode, pegs, ropes,
  os, osproc, times, cstrutils
```
without using any for the sole purpose of exporting symbols

alternatives:
* add compiler support for `discard modulename` (could help in other situations too) to disable at import-site (as opposed to {.used.} which is at importee-site)
* amend --app:lib to only apply for sfIsMainModule

Reply:
Modules that have procs that use ``.exportc, dynlib`` (or maybe just ``.export``)  should instead be considered as "used".

Reply:
excellent, yes

Reply:
I've removed this as your idea is better, can be done in a separate PR